lib/kennel.rb

```diff
@@ -109,13 +111,13 @@ def generated
             kennel_id = project.kennel_id
             if project_filter
               known << kennel_id
-              next [] if kennel_id != project_filter
+              next [] unless project_filter.include?(kennel_id)
             end
             project.validated_parts
           end
 
           if project_filter && parts.empty?
-            raise "#{project_filter} does not match any projects, try any of these:\n#{known.uniq.sort.join("\n")}"
+            raise "#{project_filter_raw} does not match any projects, try any of these:\n#{known.uniq.sort.join("\n")}"
```

Comment:
maybe better to use a single interface throughout:
```suggestion
            raise "#{project_filter.join(",")} does not match any projects, try any of these:\n#{known.uniq.sort.join("\n")}"
```

Reply:
It's not clear to me that either one is better than the other. Advantage of using `_raw`: encapsulation of the format - no need to ensure that this `join` is the reverse of the `split` elsewhere. Advantage of _not_ using `_raw`: smaller interface.

🤷‍♀️